wodles/aws/aws_s3.py

```diff
@@ -574,44 +568,21 @@ def _get_last_key_processed(self, aws_account_id: str) -> str or None:
             A str with the key of the last file processed, None if no file has been processed yet.
         """
         query_last_key = self.db_connector.execute(
-            self.sql_find_last_key_processed.format(table_name=self.db_table_name, bucket_path=self.bucket_path,
-                                                    aws_account_id=aws_account_id, prefix=self.prefix))
+            self.sql_find_last_key_processed.format(table_name=self.db_table_name), {'bucket_path': self.bucket_path,
+                                                                                     'aws_account_id': aws_account_id,
+                                                                                     'prefix': f'{self.prefix}%'})
         try:
             return query_last_key.fetchone()[0]
         except (TypeError, IndexError):
             # if DB is empty for a region
             return None
 
-    def _get_formatted_last_key_query(self, aws_region: str, aws_account_id: str) -> str:
-        """
-        Return the query used to fetch the last log key processed from the database formatted.
-
-        Parameters
-        ----------
-        aws_region : str
-            The region of the bucket.
-        aws_account_id : str
-            The AWS account ID used to collect the logs.
-
-        Returns
-        -------
-        str
-            The query formatted depending on the object's attributes.
-        """
-        return self.sql_find_last_key_processed.format(bucket_path=self.bucket_path,
-                                                        table_name=self.db_table_name,
-                                                        aws_region=aws_region,
-                                                        prefix=self.prefix,
-                                                        aws_account_id=aws_account_id)
-
     def already_processed(self, downloaded_file, aws_account_id, aws_region):
-        cursor = self.db_connector.execute(self.sql_already_processed.format(
-            bucket_path=self.bucket_path,
-            table_name=self.db_table_name,
-            aws_account_id=aws_account_id,
-            aws_region=aws_region,
-            log_name=downloaded_file
-        ))
+        cursor = self.db_connector.execute(self.sql_already_processed.format(table_name=self.db_table_name), {
```

Comment:
This `self.db_table_name` attribute is defined in the child classes but not in this one, which causes warnings. Since some methods use it, it might be useful to define it in `__init__` of this parent class as well, as `None`:
```
self.db_table_name = None
```

Reply:
This is already being fixed in #13389 which is where we are upgrading, improving and fixing gcp's Unit tests as well as applying any required change related to those tests, such as this one